pkg/skaffold/build/sequence.go

```diff
@@ -39,22 +39,32 @@ func InSequence(ctx context.Context, out io.Writer, tags tag.ImageTags, artifact
 
 		tag, present := tags[artifact.ImageName]
 		if !present {
-			return nil, fmt.Errorf("unable to find tag for image %s", artifact.ImageName)
+			results = append(results, Result{
```

Comment:
Before, build.InSequence would return once it encounters an error.
With this change, it will continue building other artifacts even if one failed. 

Does this desired behavior? or was previous behavior intended behavior?

The only problem i see with this is, initially we were failing fast. 
For a skaffold user, say i am running `skaffold build` on a skaffold file with 4 artifacts, 
They get built in seq. First Aritfact fails, i don't get to see the failure untill all build finish.
I make a change, which still does not fix the error and it keeps building the rest of artifacts.

On the other hand, the argument to add support this is, the developer can see all the errors and then fix all before running the next `skaffold build`/ `skaffold dev` loop.

I think this is something to note either in release notes or some other place appropriate.

Reply:
Great point. I thought about this, and not completely sure that it is fine. 

If there are dependencies, then this change might break those users, as the downstream dependencies are going to be built against a stale or missing image potentially. But that is not a use case we want to explicitly support, so let's ignore this. 

If there is no dependency between artifacts, then letting the other artifacts build should be fine and can actually gain productivity for the user as while one artifact is broken, and the user can start to fix it, the others can be built in the background. However this assumes that the user can "catch" the error in the streaming logs which - as you point it out - becomes hard when in `skaffold dev` there are lots of other artifacts. And with the Event API becoming "batch" instead of "streaming" with this change - the user won't be able to find the information that one of 10 artifacts is broken.

Reply:
I did this intentionally in this change, because I think the previous behavior was wrong (see the last part of the PR description).

I don't think we should be encouraging developers to have implicit dependencies between their build artifacts. if we want to do this in a separate change by making it *explicit* (through some DAG that we enable users to specify in their skaffold.yaml or something) that's fine, but it seems weird to hide that behavior like we were doing before.

we can add this as part of the release notes when this goes in.

Reply:
@balint responded to this without refreshing to pick up your comment. I understand your point, it seems like a fringe case to me. do we really think that people aren't going to be able to catch build errors because there are too many logs? also, with the event API the user will be able to query the state and quickly see which artifact is broken: the build status will be `Failed`, and they can check the event log for the error.

we could consider adding a `--fail-fast` flag to dev/build if you feel strongly about this. WDYT?